modules/collections/client/controllers/collections.controllers.js

```diff
@@ -0,0 +1,344 @@
+          .then(reloadEdit)
+          .catch(function(/* res */) {
+            $scope.showError('Could not remove document from "'+ $scope.collection.displayName +'".', [], reloadEdit, 'Remove Document Error');
+          });
+      };
+
+      $scope.delete = function() {
+        var modalView = $uibModal.open({
+          animation: true,
+          templateUrl: 'modules/utils/client/views/partials/modal-confirm-delete.html',
+          controller: function($scope, $state, $uibModalInstance) {
+            var self = this;
+            self.dialogTitle = "Delete Collection";
+            self.name = $scope.collection.displayName;
+            self.ok = function() {
+              $uibModalInstance.close($scope.collection);
+            };
+            self.cancel = function() {
+              $uibModalInstance.dismiss('cancel');
+            };
+          },
+          controllerAs: 'self',
+          scope: $scope,
+          size: 'md'
+        });
+        modalView.result.then(function(/* res */) {
+          CollectionModel.deleteId($scope.collection._id)
+            .then(function(/* res */) {
+              // deleted show the message, and go to list...
+              $scope.showSuccess('"'+ $scope.collection.displayName +'"' + ' was deleted successfully.', goToList, 'Delete Success');
+            })
+            .catch(function(/* res */) {
+              // could have errors from a delete check...
+              $scope.showError('"'+ $scope.collection.displayName +'"' + ' was not deleted.', [], reloadEdit, 'Delete Error');
+            });
+        });
+      };
+
+      $scope.confirmPublishView = function(isPublishing) {
+        return $uibModal.open({
+          animation: true,
+          templateUrl: 'modules/utils/client/views/partials/modal-confirm-generic.html',
+          controller: function($scope, $state, $uibModalInstance) {
+            var self = this;
+            self.title = isPublishing ? 'Publish Collection?' : 'Unpublish Collection?';
+            self.question = 'Are you sure you want to ' + (isPublishing ? 'publish "' : 'unpublish "') + $scope.collection.displayName + '"?';
+            self.actionOK = isPublishing ? 'Publish' : 'Unpublish';
+            self.actionCancel = 'Cancel';
+            self.ok = function() {
+              $uibModalInstance.close($scope.project);
+            };
+            self.cancel = function() {
+              $uibModalInstance.dismiss('cancel');
+            };
+          },
+          controllerAs: 'self',
+          scope: $scope,
+          size: 'md',
+          windowClass: 'modal-alert',
+          backdropClass: 'modal-alert-backdrop'
+        });
+      };
+
+      $scope.publish = function() {
+        var publishedMainDocs = _.filter($scope.collection.mainDocuments, function (item) {
+          return item.document.isPublished === true;
+        });
+
+        if (publishedMainDocs.length < 1) {
```

Comment:
Added validation to ensure at least one published main document is present when publishing the collection.